add option "default" for dropDown choices. Closes #3529

basters opened the pull request:
## What does this PR do?
It adds `default: boolean` field for `DropDown` choices.
In case when `default: true` option was specified for one of `DropDown` choices it will update default shown text

Also was changed a bit approach for testing this component.
Currently we are able to test life cycle hook and inputs based things.

### Associated Issue
Fixes #3529

RCopeland reviewed (approved):
lgtm

afebbraro commented on angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.interfaces.ts:
```diff
@@ -51,6 +51,11 @@ export interface ISprkDropdownChoice {
+   * If `default` is set to `true` and `dropdownType` is set to `informational`
+   * then `triggerText` will be automatically changed to this choice `value`
+   */
+  default?: boolean;
```
```suggestion
  isDefault?: boolean;
``` 
Could you update this?

basters replied:
Will do, thanks! Could you also check my comment in [related issue](https://github.com/sparkdesignsystem/spark-design-system/issues/3529#issuecomment-704789437)?

afebbraro reviewed (changes requested):
Just left a comment! Thanks!

basters commented:
Fixed! And also changed a bit logic for "active" elements and isDefault. Now if you have isDefault: true that choice always will be marked as "active".

RCopeland reviewed (changes requested):
found a typo and a small issue in the tests, otherwise lgtm
